pypi_push.sh

```diff
@@ -0,0 +1,25 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# first bump up package.json manually, commit and tag
+rm airflow/www_rbac/static/dist/*
+cd airflow/www_rbac/static/
+yarn
+npm run build
+cd ../../..
+python setup.py sdist upload
```

Comment:
Having this in a script is a good first step - but this is not enough for us. This needs to be integrated into `setup.py` so that running `sdist` does these steps - the "official" releases of Apache projects are onto dist.apache.org. Also having it run directly as part of setup.py means it can't be forgotten.

Reply:
This script generates the frontend build, make an Airflow build and tries to upload it to pypi. So instead of running the `python setup.py sdist upload` separately, we can simply use this script as an automated script to upload the new release. Thoughts? 
P.S: I followed the same pattern as [`incubator-superset`](https://github.com/apache/incubator-superset/blob/master/pypi_push.sh).

Reply:
The problem is there are other uploads than to PyPI.

Reply:
Something like https://stackoverflow.com/questions/11331175/hook-to-add-commands-to-distutils-build is what I think we should do - and probably for `build`, which is a step of `sdist` I think.

Reply:
If I override the main setup() function, and do the whole build process in that function, that would have other side effects. 
Users installing airflow using `pip install ..` would also run this command in that case, which is not required in any way. (and they'd have to install `npm` on their local systems as well)
The whole purpose of this build process is to make a ready to use build, with all processed Javascript and CSS files. For the initial release, can we do it with this script? as the PR is already pretty huge, and then we can create a new issue on JIRA to make this efficient.

Reply:
Having working assets from a `pip install .` sounds like a benefit to me. Though over-ridding `setup()` would be that running other commands (such as `setup.py --version` would take a long time, so that should be avoided)



I'm just checking something in the python sources, but you might be able to add a sub-command to build, or sdist that would do what we need without the over-load problems. Give me a moment.

Reply:
I think overriding the `build` command will work - they use https://github.com/python/cpython/blob/6f0eb93183519024cb360162bdd81b9faec97ba6/Lib/distutils/cmd.py#L292-L300 `get_finalized_command()` to call sub-commands, meaning it would call our build command, not the built in one.

Maybe.

A custom command in setup.py so we can do `setup.py compile_assets sdist` might be the right answer for now. I've not done a release myself yet so I don't know where the instructions on it are.

@bolkedebruin would changing the release to add an extra command before creating the sdist/bdist be okay? Where are the instructions so we can update them? Is it the wiki?

Reply:
Of course it is. @ashb

Reply:
Wiki indeed